skyportal/facility_apis/sedmv2.py

```diff
@@ -0,0 +1,251 @@
+
+    if request.payload["exposure_time"] < 0:
+        raise ValueError('exposure_time must be positive.')
+
+    if request.payload["exposure_counts"] < 1:
+        raise ValueError('exposure_counts must be at least 1.')
+
+    if request.payload["maximum_airmass"] < 1:
+        raise ValueError('maximum_airmass must be at least 1.')
+
+    if (
+        request.payload["minimum_lunar_distance"] < 0
+        or request.payload["minimum_lunar_distance"] > 180
+    ):
+        raise ValueError('maximum_airmass must be within 0-180.')
+
+    if request.payload["priority"] < 1 or request.payload["priority"] > 5:
+        raise ValueError('priority must be within 1-5.')
+
+
+class SEDMV2API(FollowUpAPI):
+    """SkyPortal interface to the Spectral Energy Distribution machine (SEDMv2)."""
+
+    @staticmethod
+    def submit(request):
+        """Submit a follow-up request to SEDMv2.
+
+        Parameters
+        ----------
+        request: skyportal.models.FollowupRequest
+            The request to submit.
+        """
+
+        from ..models import FacilityTransaction, DBSession
+
+        validate_request_to_sedmv2(request)
+
+        if cfg['app.sedmv2_endpoint'] is not None:
+            altdata = request.allocation.altdata
+
+            if not altdata:
+                raise ValueError('Missing allocation information.')
+
+            payload = {
+                'obj_id': request.obj_id,
+                'allocation_id': request.allocation.id,
+                'payload': request.payload,
+            }
+
+            r = requests.post(
+                cfg['app.sedmv2_endpoint'],
+                json=payload,
+                headers={"Authorization": f"token {altdata['api_token']}"},
+            )
+
+            if r.status_code == 200:
+                request.status = 'submitted'
+            else:
+                request.status = f'rejected: {r.content}'
+
+            transaction = FacilityTransaction(
+                request=http.serialize_requests_request(r.request),
+                response=http.serialize_requests_response(r),
+                followup_request=request,
+                initiator_id=request.last_modified_by_id,
+            )
+        else:
+            request.status = 'submitted'
+
+            transaction = FacilityTransaction(
+                request=None,
+                response=None,
+                followup_request=request,
+                initiator_id=request.last_modified_by_id,
+            )
+
+        DBSession().add(transaction)
+
+    @staticmethod
+    def delete(request):
+        """Delete a follow-up request from SEDMv2 queue.
+
+        Parameters
+        ----------
+        request: skyportal.models.FollowupRequest
+            The request to delete from the queue and the SkyPortal database.
+        """
+
+        from ..models import DBSession, FollowupRequest, FacilityTransaction
+
+        validate_request_to_sedmv2(request)
```

Comment:
Do we need to submit all the required parameters even though we are just deleting a followup?

Reply:
Great point. Removed.